### ISSUE TITLE
Segmentation fault error

### ISSUE THREAD

**Issue by impulsecorp** [titu1994/progressive-neural-architecture-search#2]
No matter what I do, I get a "Segmentation fault (core dumped)" when it tries the 2nd model. For example, I tried adding CUDA_VISIBLE_DEVICES=0 and CUDA_VISIBLE_DEVICES=-1 but that did not help. I am using a totally new AWS server just for this program, and I tried both the Ubuntu and Amazon Linux Deep Learning AMIs (such as https://aws.amazon.com/marketplace/pp/B077GCZ4GR ). I pip installed Tensorflow-GPU and Keras.  

I will use whatever server configuration is needed to get this working. Everything runs fine for the first model, see below:

Model #1 / #36
Predicted actions :  [0, '3x3 dconv', 0, '3x3 dconv']
2018-06-25 15:13:07.805285: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2018-06-25 15:13:07.805332: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-06-25 15:13:07.805347: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-06-25 15:13:07.805359: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-06-25 15:13:07.805497: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10763 MB memory) -> physical GPU (device: 0, name: Tesla K80, pci bus id: 0000:00:1e.0, compute capability: 3.7)
Train on 50000 samples, validate on 10000 samples
Epoch 1/1
50000/50000 [==============================] - 10s 201us/step - loss: 1.8916 - acc: 0.3220 - val_loss: 1.8114 - val_acc: 0.3520

Epoch 00001: val_acc improved from -inf to 0.35200, saving model to weights/temp_network.h5
10000/10000 [==============================] - 0s 37us/step

Manager: Accuracy =  0.352
Final Accuracy :  0.352

Finished 1 out of 36 models !

Actions :
inputs :  [(0, 1.0), (1, 0.0), (2, 0.0), (3, 0.0), (4, 0.0)]
ops :  [('3x3 dconv', 1.0), ('5x5 dconv', 0.0), ('7x7 dconv', 0.0), ('1x7-7x1 conv', 0.0), ('3x3 maxpool', 0.0), ('3x3 avgpool', 0.0)]
inputs :  [(0, 1.0), (1, 0.0), (2, 0.0), (3, 0.0), (4, 0.0)]
ops :  [('3x3 dconv', 0.0), ('5x5 dconv', 2.0), ('7x7 dconv', 0.0), ('1x7-7x1 conv', 0.0), ('3x3 maxpool', 0.0), ('3x3 avgpool', 0.0)]

Model #2 / #36
Predicted actions :  [0, '3x3 dconv', 0, '5x5 dconv']
2018-06-25 15:13:19.862650: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2018-06-25 15:13:19.862695: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-06-25 15:13:19.862710: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2018-06-25 15:13:19.862717: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2018-06-25 15:13:19.862840: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10763 MB memory) -> physical GPU (device: 0, name: Tesla K80, pci bus id: 0000:00:1e.0, compute capability: 3.7)
Segmentation fault (core dumped)
ubuntu@ip-172-31-17-27:~/pnas$ ^C

**Comment by titu1994** [comment]
Keras 2.2.0 and TF 1.8 Seg fault when you use K.clear_session() and then fit another model. 

I don't know any current solution other than downgrading Keras to 2.1.6 or TF to 1.7

**Comment by impulsecorp** [comment]
Yes, that fixed it!
